src/connection-manager/amendments.ts

```diff
@@ -0,0 +1,141 @@
+import axios from 'axios'
+import createHash from 'create-hash'
+
+import { query } from '../shared/database/utils'
+import { AmendmentsInfo } from '../shared/types'
+import logger from '../shared/utils/logger'
+
+const log = logger({ name: 'amendments' })
+
+const cachedAmendmentIDs = new Map<
+  string,
+  { name: string; deprecated: boolean }
+>()
+const cachedRippledVersions = new Map<string, string>()
+
+const ACTIVE_AMENDMENT_REGEX =
+  /^\s*REGISTER_F[A-Z]+\s*\((?<amendmendName>\S+),\s*.*$/u
+const RETIRED_AMENDMENT_REGEX =
+  /^ .*retireFeature\("(?<amendmendName>\S+)"\)[,;].*$/u
+
+const AMENDMENT_VERSION_REGEX =
+  /\| \[(?<amendmentName>[a-zA-Z0-9_]+)\][^\n]+\| (?<version>v[0-9]*\.[0-9]*\.[0-9]*|TBD) *\|/u
+
+/**
+ * Fetch a list of amendments names from rippled file.
+ *
+ * @returns The list of amendment names.
+ */
+async function fetchAmendmentNames(): Promise<Map<string, boolean> | null> {
```

Comment:
This can probably be cleaned up once https://github.com/XRPLF/rippled/pull/4781 is merged

Reply:
Nice feature! I will add a TODO